src/components/input/DurationInput.js

```diff
@@ -0,0 +1,404 @@
+import React, { PureComponent } from 'react';
+import PropTypes from 'prop-types';
+import { IconChevronUpSmallOutline, IconChevronDownSmallOutline } from '@teamleader/ui-icons';
+import Icon from '../icon';
+import cx from 'classnames';
+import InputBase from './InputBase';
+import { TextBody } from '../typography';
+import Box, { pickBoxProps } from '../box';
+import ValidationText from '../validationText';
+import { durationToSeconds } from '../utils';
+
+import theme from './theme.css';
+
+class SpinnerControls extends PureComponent {
+  render() {
+    const { inverse, spinnerUpProps, spinnerDownProps } = this.props;
+    const iconProps = {
+      color: inverse ? 'teal' : 'neutral',
+      element: 'button',
+      tabIndex: '-1',
+      tint: inverse ? 'lightest' : 'darkest',
+      type: 'button',
+    };
+
+    return (
+      <div className={theme['spinner']}>
+        <Icon className={theme['spinner-up']} {...spinnerUpProps} {...iconProps}>
+          <IconChevronUpSmallOutline />
+        </Icon>
+        <Icon className={theme['spinner-down']} {...spinnerDownProps} {...iconProps}>
+          <IconChevronDownSmallOutline />
+        </Icon>
+      </div>
+    );
+  }
+}
+
+class DurationInput extends PureComponent {
+  state = {
+    hours: 0,
+    minutes: 0,
+    seconds: 0,
+    hoursInputHasfocus: false,
+    minutesInputHasfocus: false,
+    secondsInputHasfocus: false,
+  };
+
+  updateStep = (type, n) => {
+    const { hours, minutes, seconds } = this.state;
+    let newHours = hours;
+    let newMinutes = minutes;
+    let newSeconds = seconds;
+
+    switch (type) {
+      case 'hours':
+        if (hours >= 0 && n >= 0) {
+          newHours = hours + n;
+        } else if (hours >= Math.abs(n) && n < 0) {
+          newHours = hours + n;
+        } else if (hours <= Math.abs(n) && n < 0) {
+          newHours = 0;
+        }
+        this.setState({ hours: newHours }, () => this.props.onChange(durationToSeconds(newHours, minutes, seconds)));
+        break;
+
+      case 'minutes':
```

Comment:
I think you can make all these calculations a lot simpler by using some Math functions and operators.